src/postamble.js

```diff
@@ -192,6 +185,10 @@ function run() {
   }
 #endif
 
+#if STACK_OVERFLOW_CHECK
+  stackCheckInit();
```

Comment:
This code both moved and lost the condition `if (!pthread)` - do we not need that?

Reply:
We already have early return for `if (ENVIRONMENT_IS_PTHREAD)` and `if (ENVIRONMENT_IS_WASM_WORKER)` so by the time we get here know were the main runtime thread.